### DISCUSSION TITLE
Add support for OKLCH notation for declaring colors

### DISCUSSION THREAD

**Discussion by NNBnh** [godotengine/godot-proposals#8844] [Engine Core]
From [OKLCH in CSS: why we moved from RGB and HSL](https://evilmartians.com/chronicles/oklch-in-css-why-quit-rgb-hsl):

> OKLCH is a new way to define colors. Each item corresponds as follows:
> 
> - `L` is perceived lightness (0%-100%). "Perceived" means that it has consistent lightness for our eyes, unlike L in `hsl`.
> - `C` is chroma, from gray to the most saturated color.
> - `H` is the hue angle (0-360).
> 
> The benefits of OKLCH:
> 
> 1. OKLCH frees designers from the need to manually choose every color. They can define a formula, choose a few colors, and an entire design system **palette is automatically generated**.
> 2. OKLCH can be used for wide-gamut **P3 colors**. For instance, new devices (like those from Apple) can display more colors than old sRGB monitors, and we can use OKLCH to specify these new colors.
> 3. Unlike `hsl`, OKLCH is **better for color modifications** and palette generation. It uses perceptual lightness, so no more unexpected results, like we had with `darken` in Sass.
> 4. Further, with its predictable lightness, OKLCH provides **better accessibility**.
> 5. Unlike `rgb` or `hex` (`#ca0000`), OKLCH is **human readable**. You can quickly and easily know which color an OKLCH value represents simply by looking at the numbers. OKLCH works like HSL, but it encodes lightness better than HSL.

Useful link: [OKLCH Color Picker & Converter](https://oklch.com)

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot-proposals/issues/4353 and https://github.com/godotengine/godot-proposals/issues/7954.

Note that Godot 4.0 and later already has OKHSL support, which mitigates many of the issues found in the RGB and HSL color schemes. This is also supported in Gradient interpolation.
